Add installation guide (#68)
neuroinformatics-unit/poseinterface: docs/source/installation.md

(target-installation)=
# Installation

## Create a virtual environment

We recommend installing `poseinterface` in a virtual environment.
The instructions below use [uv](https://docs.astral.sh/uv/), but feel free to use any virtual-environment tool you are familiar with, including [conda](https://docs.conda.io/).

In your working directory, create and activate a new environment:

```sh

> uv venv --python=3.13

Reviewer: ```suggestion
uv venv
```
To avoid having to update this every time we drop/add support for different python versions, I would omit it here and let uv select the version that satisfies our constraint (currently >=3.12) in pyproject.toml.

Author: Would be good to avoid updating, but I do think we have to include a version.

I looked into this. This guide is for users installing from PyPI, so there's usually no `pyproject.toml` in the working directory for `uv venv` to read `requires-python` from; it just takes whatever python3 is on PATH (though that's configurable via `UV_PYTHON_PREFERENCE`). On my machine that's conda base env's 3.11, i.e. below our floor. Worse, `uv pip install poseinterface` on 3.11 doesn't error, it silently resolves to the older 0.0.1 stub.

So, to be on the safe side, I would specify a Python version, but perhaps that should be 3.14 (more future-proof)?

Alternatively, we could do `uv venv --python>"=3.12"`, which is the safest of all, in the sense that it keeps the precise minor version flexible.

Author: We could also yank the placeholder 0.0.1 release from PyPI to avoid silent downgrades.

Author: > Alternatively, we could do uv venv --python>"=3.12", which is the safest of all, in the sense that it keeps the precise minor version flexible.

I went with this for now, let me know if you have a different preference.

Author: > We could also yank the placeholder 0.0.1 release from PyPI to avoid silent downgrades.

Done.

source .venv/bin/activate  # On macOS and Linux
.venv\Scripts\activate     # On Windows PowerShell
```

## Install the package

> With your environment activated, install `poseinterface` from PyPI:

Reviewer: ```suggestion
With your environment activated, install the latest release of `poseinterface` from PyPI:
```

```sh
uv pip install poseinterface
```

> Or directly from the `main` branch on GitHub:

Reviewer: ```suggestion
Or, to get the latest development version from the `main` branch on GitHub:
```

```sh
uv pip install git+https://github.com/neuroinformatics-unit/poseinterface.git@main
```

`uv` may be omitted from these commands if it is not available;
`pip install ...` works in any standard virtual environment.

## Update the package

To update to the latest release:

```sh
uv pip install -U poseinterface
```

:::{admonition} For developers
:class: tip

If you would like to contribute to `poseinterface`, see the

> [contributing guide](target-contributing) for developer setup instructions.

Reviewer: ```suggestion
[contributing guide](target-contributing) for developer setup instructions
and coding guidelines.
```
copying [movement](https://movement.neuroinformatics.dev/latest/user_guide/installation.html#install-the-package)?
